C++ API parity: PReLU

pbelevich opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27309 C++ API parity: MultiheadAttention
* #27382 C++ API parity: Linear
* #27538 C++ API parity: Threshold
* #27537 C++ API parity: Tanhshrink
* #27536 C++ API parity: Tanh
* #27535 C++ API parity: Softsign
* #27534 C++ API parity: Softshrink
* #27489 C++ API parity: Softplus
* #27488 C++ API parity: Sigmoid
* #27487 C++ API parity: CELU
* #27437 C++ API parity: RReLU
* #27436 C++ API parity: ReLU6
* #27435 C++ API parity: ReLU
* **#27429 C++ API parity: PReLU**

Differential Revision: [D17835412](https://our.internmc.facebook.com/intern/diff/D17835412)

pbelevich committed:
C++ API parity: PReLU

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: PReLU"




[ghstack-poisoned]

yf225 commented on torch/csrc/api/include/torch/nn/options/activation.h:
```diff
@@ -56,5 +57,19 @@ struct LeakyReLUOptions {
+struct PReLUOptions {
+  PReLUOptions() {}
+
+  /// number of a to learn. Although it takes an int as input, there is only
```
nit:
```suggestion
  /// number of `a` to learn. Although it takes an int as input, there is only
```

yf225 commented on torch/csrc/api/include/torch/nn/options/activation.h:
```diff
@@ -56,5 +57,19 @@ struct LeakyReLUOptions {
+  /// two values are legitimate: 1, or the number of channels at input. Default: 1
+  TORCH_ARG(int64_t, num_parameters) = 1;
+
+  /// the initial value of a. Default: 0.25
```
nit:
```suggestion
  /// the initial value of `a`. Default: 0.25
```

yf225 commented on torch/csrc/api/include/torch/nn/options/activation.h:
```diff
@@ -56,5 +57,19 @@ struct LeakyReLUOptions {
+
+/// Options for PReLU functional and module.
+struct PReLUOptions {
+  PReLUOptions() {}
```
We probably don't need to declare the default constructor as it should be provided automatically.

yf225 reviewed (approved):
@pbelevich Thanks a lot for the great work! It looks great overall, and I left some very minor comments.

yf225 commented on torch/csrc/api/include/torch/nn/modules/activation.h:
```diff
@@ -120,5 +120,31 @@ class TORCH_API LogSigmoidImpl : public torch::nn::Cloneable<LogSigmoidImpl> {
+class TORCH_API PReLUImpl : public torch::nn::Cloneable<PReLUImpl> {
+ public:
+  PReLUImpl() : PReLUImpl(PReLUOptions()) {}
+  explicit PReLUImpl(const PReLUOptions& options_);
```
Would we be able to combine the above two constructors and just have the following?
```cpp
  explicit PReLUImpl(const PReLUOptions& options_ = {});
```

pbelevich committed:
Update on "C++ API parity: PReLU"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: PReLU"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: PReLU"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: PReLU"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: PReLU"


Differential Revision: [D17835412](https://our.internmc.facebook.com/intern/diff/D17835412)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: PReLU"


Differential Revision: [D17835412](https://our.internmc.facebook.com/intern/diff/D17835412)

[ghstack-poisoned]
